### PR TITLE
feat(Airdrops): Component for selecting recipient addresses and members

### PR DESCRIPTION
**Pull request by micieslak**
### What does the PR do

Implements selector component for adding eth addresses and eth members.

Closes: #9799

### Affected areas
only new components in Storybook

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Screenshot 2023-04-03 10:32:51
Screenshot 2023-04-03 10:33:01

https://user-images.githubusercontent.com/20650004/229463568-ab3af29d-817e-47b0-b6ef-9e21fe095f49.mp4

### CODE REVIEW

**Comment by caybro** [comment]
@micieslak still WIP? some screenshot/cool video? ;)

File: ui/app/AppLayouts/Chat/controls/community/AddressesInputList.qml [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            rightPadding: 0
+
+            multiline: true
+
+            height: Math.min(root.maximumTextInputHeight,
+                             edit.implicitHeight + topPadding + bottomPadding)
+
+            placeholderText: qsTr("Example: 0x39cf...fbd2")
+
+            Keys.onReturnPressed: {
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/controls/community/AddressesInputList.qml:153]: What about `onEnterPressed`? ;)

File: ui/app/AppLayouts/Chat/controls/community/AddressesSelectorPanel.qml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                font.pixelSize: Theme.tertiaryTextFontSize
+                elide: Text.ElideRight
+            }
+
+            Item { Layout.fillWidth: true }
+
+            StatusBaseText {
+                visible: !root.loading && root.validAddressesCount > 0
+                color: Theme.palette.baseColor1
+                text: qsTr("%n valid address(es)", "", root.validAddressesCount)
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/controls/community/AddressesSelectorPanel.qml:51]: Must be `"%n valid address(s)"`... yes

File: ui/app/AppLayouts/Chat/controls/community/AddressesSelectorPanel.qml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            }
+
+            StatusBaseText {
+                visible: !root.loading && root.invalidAddressesCount > 0
+                color: Theme.palette.dangerColor1
+                text: root.validAddressesCount > 0
+                      ? qsTr("%n invalid",
+                             "invalid addresses, where \"addresses\" is implicit",
+                             root.invalidAddressesCount)
+                      : qsTr("%n invalid address(es)", "", root.invalidAddressesCount)
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/controls/community/AddressesSelectorPanel.qml:63]: Same here

**Comment by alexjba** [comment]
Very nice work! The code LGTM 👍 

A few things I noticed:
1. The linked GH issue is wrong`Closes: https://github.com/status-im/status-desktop/pull/9955` in the description.
2. In Storybook we can input multiple addresses at the same time. Either separated by space or new line. Will it be the same in the app, or we're missing a few restrictions in the component? (BTW, I like this feature 😄) - see video
3. The `MembersSelectorPanel` is using a listview and it's scrollable by default, but the `AddressesInputList` is just a repeater (no scroll support). Is there a reason why the `AddressesInputList` is not scrollable by default? If it's not scrollable it can grow outside the screen (print screen below) and we'll probably wrap it inside a Flickable anyway.

https://user-images.githubusercontent.com/47811206/229348436-cf91b30c-1aba-46a9-b871-f883ad6039e0.mov

Screenshot 2023-04-02 at 13 53 17

File: ui/app/AppLayouts/Chat/controls/community/AddressesInputList.qml (hunk 1 above)
Comment (micieslak) [reply ui/app/AppLayouts/Chat/controls/community/AddressesInputList.qml:184]: `onEnterPressed` seems to be problematic here bc of `StatusBaseInput` internal details. `onReturnPressed` just works as expected...

**Comment by micieslak** [comment]
Thanks @alexjba for review :)

> 1. The linked GH issue is wrong`Closes: https://github.com/status-im/status-desktop/pull/9955` in the description.
Right, it's wrong, fixed :)

> 2. In Storybook we can input multiple addresses at the same time. Either separated by space or new line. Will it be the same in the app, or we're missing a few restrictions in the component? (BTW, I like this feature smile) - see video
I'm not sure what you mean. Parsing of the input is not part of the component. It's intended to be done externally for easier integration. It's not clear now how it will be parsed (in the backend or us side).
> 3. The `MembersSelectorPanel` is using a listview and it's scrollable by default, but the `AddressesInputList` is just a repeater (no scroll support). Is there a reason why the `AddressesInputList` is not scrollable by default? If it's not scrollable it can grow outside the screen (print screen below) and we'll probably wrap it inside a Flickable anyway.
Already fixed. For that reason it was marked as draft :)

**Review by alexjba — APPROVED** [review]
LGTM! Great work! 👍

File: ui/app/AppLayouts/Chat/controls/community/AddressesInputList.qml (hunk 1 above)
Comment (caybro) [reply ui/app/AppLayouts/Chat/controls/community/AddressesInputList.qml:184]: Yeah, for the regular "Return" key sure, have you tried typing the number on a numpad and pressing the numpad "Enter"? ;)

Comment (micieslak) [reply ui/app/AppLayouts/Chat/controls/community/AddressesInputList.qml]: Fixed! Thanks for pointing that :)

**Review by caybro — APPROVED** [review]
LGTM!

**Pull request merged**